app/decorators/article_decorator.rb

```diff
@@ -10,8 +10,18 @@ def user_data_info_to_json
     DataInfo.to_json(object: cached_user, class_name: "User", id: user_id, style: "full")
   end
 
+  def current_state
```

Comment:
Wonder what your thoughts are regarding using the `ActiveSupport::StringInquirer`? (https://api.rubyonrails.org/classes/ActiveSupport/StringInquirer.html)

We could have the following:

```ruby
def current_state
  state = if !published?
    "unpublished"
  elsif scheduled?
    "scheduled"
  else
    "published"
  end
  ActiveSupport::StringInquirer.new(state)
end
```